### PR TITLE
[nginx-ingress-controller] Enable configuration to disable http2

### LINKED ISSUE #960: Nginx Ingress Controller - Safari 9 craps out with http2
I've literally wasted hours upon hours tracking this down. :-)

Using the nginx controller with SSL, Safari 9 will refuse to load any site. I'm sure this is a bug on Safari's part but I finally tracked it down to http2 in the listen 443 ssl line. I had to create a custom template to remove the http2 lines in order to get Safari (mobile or mac) to load our sites. None of the other browsers including older Safari browsers had an issue. I think http2 support was added in recent Safari version on the latest OS but obviously something is not right. Unfortunately, it wasn't just my local mac safari either as we had customers all over the world complain they couldn't access the sites with safari! 

Maybe this should be a setting? Or if there's a known workaround I couldn't figure out? 

https://github.com/kubernetes/contrib/blob/master/ingress/controllers/nginx/nginx.tmpl#L157

### PR DESCRIPTION
**Pull request by aledbf**
fixes #960

Latest safari have issues with http2 https://trac.nginx.org/nginx/ticket/979

### CODE REVIEW

**Comment by aledbf** [comment]
ping @bprashanth

File: ingress/controllers/nginx/nginx/main.go [hunk 1]
```diff
@@ -214,6 +214,11 @@ type nginxConfiguration struct {
 	// http://nginx.org/en/docs/http/ngx_http_gzip_module.html
 	UseGzip bool `structs:"use-gzip,omitempty"`
 
+	// Enables or disables the use of the nginx module that provides support for HTTP/2 in secure connections
+	// http://nginx.org/en/docs/http/ngx_http_v2_module.html
```
Comment (bprashanth) [inline ingress/controllers/nginx/nginx/main.go]: are you actually disabling the module or just not advertising http2 as a supported upgrade protocol? if the latter, please modify the comment.

Comment (aledbf) [inline ingress/controllers/nginx/nginx/main.go]: done

**Comment by bprashanth** [comment]
lgtm, but you have merge conflicts

**Comment by aledbf** [comment]
done

**Comment by bprashanth** [comment]
LGTM

**Pull request merged**